### PR TITLE
Add constraint list to docs landing page

### PR DESCRIPTION
**Pull request by chrisnegus**
**1. Description of changes:**
The PR adds a list of constraints supported by Karpenter to the docs landing page. Please check that the list offers the right level of information and that the link for each constraint is useful.

**2. Does this change impact docs?**
- [X] Yes, PR includes docs updates
- [ ] Yes, issue opened: *link to issue*
- [ ] No

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: website/content/en/preview/_index.md [hunk 1]
```diff
@@ -19,11 +18,30 @@ Karpenter works by:
...
+* **Deploy workloads**: When deploying workloads, you can request that scheduling constraints be met to direct which nodes Karpenter provisions for those workloads.
+
+Karpenter supports Kubernetes scheduling constraints, including Kubernetes well-known labels, annotations, and taints.
+It also supports constraints specific to the cloud provider (such as AWS).
+The following Kubernetes and cloud provider constraints are supported by Karpenter at the provisioner, node, and pod levels:
+
+* Provisioner Level
+  - [Zones](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone)
+  - [Capacity Types]({{<ref "./provisioner" >}}),
+  - [Instance Types](https://kubernetes.io/docs/reference/labels-annotations-taints/#nodekubernetesioinstance-type)
```
Comment (tzneal) [inline website/content/en/preview/_index.md:34]: Zones, Capacity Types & Instance Types are all done via 'Requirements' which can also include other arbitrary labels.  These are just the well-known ones.

File: website/content/en/preview/_index.md [hunk 2]
```diff
@@ -19,11 +18,30 @@ Karpenter works by:
...
+  - [Taints](https://kubernetes.io/docs/concepts/scheduling-eviction/taint-and-toleration/)
+* Node Level
```
Comment (tzneal) [inline website/content/en/preview/_index.md:36]: I'm not sure this Node section belongs, they're derived from the set of all available instance types plus the requirements set on the provisioner.  They're not something separate that users can configure.

**Review by spring1843 — APPROVED** [review]
/release

File: website/content/en/preview/_index.md [hunk 3]
```diff
@@ -19,11 +18,25 @@ Karpenter works by:
...
+  - Pod affinity and anti-affinity (`podAffinity/podAntiAffinity`): Choose to run a pod on a node based on whether certain pods are running (`podAffinity`) or not running (`podAntiAffinity`) on the node. See [Inter-pod affinity and anti-affinity](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#inter-pod-affinity-and-anti-affinity) for details.
+  - Requirements (`requirements`): Choose required matching key-value pairs (such as [instance types](https://kubernetes.io/docs/reference/labels-annotations-taints/#nodekubernetesioinstance-type), [zones](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone), [computer architecture](https://kubernetes.io/docs/reference/labels-annotations-taints/#kubernetes-io-arch), and [capacity type]({{<ref "./provisioner/#capacity-type" >}}) (such as AWS spot or on-demand).
```
Comment (tzneal) [inline website/content/en/preview/_index.md:37]: Requirements aren't real things on pods, they are just a way of thinking about the the labels, affinities and tolerations.

Comment (chrisnegus) [reply website/content/en/preview/_index.md]: Fixed. Thanks!

**Pull request merged**